soc/riscv/riscv-privileged/common/soc_common.h

```diff
@@ -41,6 +41,41 @@
 void soc_interrupt_init(void);
 #endif
 
+/* RISC-V custom interrupt controller API, see CONFIG_RISCV_HAS_CUSTOM_IC */
+
+/**
+ * @brief Enable an interrupt line for RISC-V custom interrupt controller.
+ *
+ * This routine enables an interrupt line for RISC-V custom interrupt controller.
+ * Should be implemented by custom interrupt controller for 2nd level IRQs.
+ *
+ * @param irq 2nd level IRQ number to enable
+ */
+void riscv_custom_ic_irq_enable(uint32_t irq);
```

Comment:
Why not following the ARM case and move this to the `z_soc_*` namespace?

Reply:
Nice idea, didn't notice it, thanks!